src/shims/env.rs

```diff
@@ -37,15 +37,15 @@ impl EnvVars {
 }
 
 fn alloc_env_var_as_c_str<'mir, 'tcx>(
-    name: &[u8],
-    value: &[u8],
+    name: &OsStr,
+    value: &OsStr,
     ecx: &mut InterpCx<'mir, 'tcx, Evaluator<'tcx>>
 ) -> Pointer<Tag> {
-    let mut bytes = name.to_vec();
-    bytes.push(b'=');
-    bytes.extend_from_slice(value);
-    bytes.push(0);
-    ecx.alloc_os_str_as_c_str(bytes_to_os_str(bytes.as_slice()).unwrap())
+    let mut name_osstring = name.to_os_string();
+    name_osstring.push("=");
+    name_osstring.push(value);
+    name_osstring.push("\u{0000}");
+    ecx.alloc_os_str_as_c_str(name_osstring.as_os_str(), MiriMemoryKind::Env.into())
```

Comment:
`alloc_os_str_as_c_str` takes care of the trailing 0, does it not?

Reply:
> 
> 
> `alloc_os_str_as_c_str` takes care of the trailing 0, does it not?

You are right. The `null` insertion here is redundant. I will get rid of it.